flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorResourceUtils.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.configuration.TaskManagerOptions;
+import org.apache.flink.runtime.clusterframework.types.ResourceProfile;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Utility class for {@link TaskExecutorResourceSpec} of running {@link TaskExecutor}.
+ */
+public class TaskExecutorResourceUtils {
+	private static final Logger LOG = LoggerFactory.getLogger(TaskExecutorResourceUtils.class);
+
+	static final List<ConfigOption<?>> CONFIG_OPTIONS = Arrays.asList(
+		TaskManagerOptions.CPU_CORES,
+		TaskManagerOptions.TASK_HEAP_MEMORY,
+		TaskManagerOptions.TASK_OFF_HEAP_MEMORY,
+		TaskManagerOptions.NETWORK_MEMORY_MIN,
+		TaskManagerOptions.NETWORK_MEMORY_MAX,
+		TaskManagerOptions.MANAGED_MEMORY_SIZE
+	);
+
+	private static final List<ConfigOption<?>> UNUSED_CONFIG_OPTIONS = Arrays.asList(
+		TaskManagerOptions.TOTAL_PROCESS_MEMORY,
+		TaskManagerOptions.TOTAL_FLINK_MEMORY,
+		TaskManagerOptions.FRAMEWORK_HEAP_MEMORY,
+		TaskManagerOptions.FRAMEWORK_OFF_HEAP_MEMORY,
+		TaskManagerOptions.JVM_METASPACE,
+		TaskManagerOptions.JVM_OVERHEAD_MIN,
+		TaskManagerOptions.JVM_OVERHEAD_MAX,
+		TaskManagerOptions.JVM_OVERHEAD_FRACTION
+	);
+
+	static final MemorySize DEFAULT_SHUFFLE_MEMORY_SIZE = MemorySize.parse("64m");
+	static final MemorySize DEFAULT_MANAGED_MEMORY_SIZE = MemorySize.parse("128m");
+
+	private TaskExecutorResourceUtils() {}
+
+	static TaskExecutorResourceSpec resourceSpecFromConfig(Configuration config) {
+		try {
+			checkTaskExecutorResourceConfigSet(config);
+		} catch (IllegalConfigurationException e) {
+			throw new IllegalConfigurationException("Failed to create TaskExecutorResourceSpec", e);
+		}
+		return new TaskExecutorResourceSpec(
+			new CPUResource(config.getDouble(TaskManagerOptions.CPU_CORES)),
+			config.get(TaskManagerOptions.TASK_HEAP_MEMORY),
+			config.get(TaskManagerOptions.TASK_OFF_HEAP_MEMORY),
+			config.get(TaskManagerOptions.NETWORK_MEMORY_MIN),
+			config.get(TaskManagerOptions.MANAGED_MEMORY_SIZE)
+		);
+	}
+
+	private static void checkTaskExecutorResourceConfigSet(Configuration config) {
+		CONFIG_OPTIONS.forEach(option -> checkConfigOptionIsSet(config, option));
+		checkTaskExecutorNetworkConfigSet(config);
+	}
+
+	private static void checkTaskExecutorNetworkConfigSet(ReadableConfig config) {
+		if (!config.get(TaskManagerOptions.NETWORK_MEMORY_MIN).equals(config.get(TaskManagerOptions.NETWORK_MEMORY_MAX))) {
+			throw new IllegalConfigurationException(
+				"The network memory min (%s) and max (%s) mismatch, " +
+					"the network memory has to be fixed after task executor has started",
+				config.get(TaskManagerOptions.NETWORK_MEMORY_MIN),
+				config.get(TaskManagerOptions.NETWORK_MEMORY_MAX));
+		}
+	}
+
+	private static void checkConfigOptionIsSet(Configuration config, ConfigOption<?> option) {
+		if (!config.contains(option)) {
```

Comment:
Why aren't default values allowed here? I think this check is too strict in the sense that we must always set the values although we might simply take the default values of some `ConfigOptions`.